src/plugins/analysis/ip_and_uri_finder/test/test_ip_and_uri_finder.py

```diff
@@ -41,69 +40,62 @@ def city(self, address):  # pylint: disable=too-complex,inconsistent-return-stat
             raise ValueError()
 
 
-class TestAnalysisPluginIpAndUriFinder(AnalysisPluginTest):
+@pytest.fixture
+def ip_and_uri_finder_plugin(analysis_plugin):
+    analysis_plugin.reader = MockReader()
+    yield analysis_plugin
 
-    PLUGIN_NAME = 'ip_and_uri_finder'
-    PLUGIN_CLASS = AnalysisPlugin
 
-    @patch('geoip2.database.Reader', MockReader)
-    def setUp(self):
-        super().setUp()
-
-    @patch('geoip2.database.Reader', MockReader)
-    def test_process_object_ips(self):
+@pytest.mark.AnalysisPluginClass.with_args(AnalysisPlugin)
+class TestAnalysisPluginIpAndUriFinder:
+    def test_process_object_ips(self, ip_and_uri_finder_plugin):
         with tempfile.NamedTemporaryFile() as tmp:
             with open(tmp.name, 'w') as fp:
                 fp.write('1.2.3.4 abc 1.1.1.1234 abc 3. 3. 3. 3 abc 1255.255.255.255 1234:1234:abcd:abcd:1234:1234:abcd:abc'
                          'd xyz 2001:db8::8d3:: xyz 2001:db8:0:0:8d3::')
             tmp_fo = FileObject(file_path=tmp.name)
-            processed_object = self.analysis_plugin.process_object(tmp_fo)
-            results = processed_object.processed_analysis[self.PLUGIN_NAME]
-        self.assertEqual(results['uris'], [])
-        self.assertCountEqual([('1.2.3.4', '47.913, -122.3042'), ('1.1.1.123', '-37.7, 145.1833')], results['ips_v4'])
-        self.assertCountEqual([('1234:1234:abcd:abcd:1234:1234:abcd:abcd', '2.1, 2.1'), ('2001:db8:0:0:8d3::', '3.1, 3.1')],
-                              results['ips_v6'])
-
-    @patch('geoip2.database.Reader', MockReader)
-    def test_process_object_uris(self):
+            processed_object = ip_and_uri_finder_plugin.process_object(tmp_fo)
+            results = processed_object.processed_analysis[ip_and_uri_finder_plugin.NAME]
+        assert results['uris'] == []
+        assert set([('1.2.3.4', '47.913, -122.3042'), ('1.1.1.123', '-37.7, 145.1833')]) == set(results['ips_v4'])
+        assert len([('1.2.3.4', '47.913, -122.3042'), ('1.1.1.123', '-37.7, 145.1833')]) == len(results['ips_v4'])
+        assert set([('1234:1234:abcd:abcd:1234:1234:abcd:abcd', '2.1, 2.1'), ('2001:db8:0:0:8d3::', '3.1, 3.1')]), set(results['ips_v6'])
+        assert len([('1234:1234:abcd:abcd:1234:1234:abcd:abcd', '2.1, 2.1'), ('2001:db8:0:0:8d3::', '3.1, 3.1')]), len(results['ips_v6'])
```

Comment:
something went wrong here 🤔

Reply:
Did it?
`assertCountEqual` asserts that "a and b have the same elements in the same number, regardless of their order.".
"same elements [...] regardless of their" is covered by comparing the sets.
"same number" is covered by comparing the lengths (e.g. duplicate entrys would be missed without this test).